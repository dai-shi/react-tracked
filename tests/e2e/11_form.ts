import { describe, expect, it } from 'vitest';
import puppeteer from 'puppeteer';

describe('11_form', () => {
  const port = process.env.PORT || '8080';

  it('should work with recorded events', async () => {
    // eslint-disable-next-line import/no-named-as-default-member
    const browser = await puppeteer.launch();
    const page = await browser.newPage();
    await page.goto(`http://localhost:${port}/`);

    await page.waitForSelector('#app > form > div:nth-child(3) > div > input');
    await page.click('#app > form > div:nth-child(3) > div > input');
    expect(
      await page.evaluate(() => document.body.innerHTML),
    ).toMatchSnapshot();

    await page.type('#app > form > div:nth-child(3) > div > input', '1');
    expect(
      await page.evaluate(() => document.body.innerHTML),
    ).toMatchSnapshot();

    await page.type('#app > form > div:nth-child(3) > div > input', '1');
    expect(
      await page.evaluate(() => document.body.innerHTML),
    ).toMatchSnapshot();

    await page.type('#app > form > div:nth-child(3) > div > input', '1');
    expect(
      await page.evaluate(() => document.body.innerHTML),
    ).toMatchSnapshot();

    await page.type('#app > form > div:nth-child(5) > div > input', '2');
    expect(
      await page.evaluate(() => document.body.innerHTML),
    ).toMatchSnapshot();

    await page.type('#app > form > div:nth-child(5) > div > input', '2');
    expect(
      await page.evaluate(() => document.body.innerHTML),
    ).toMatchSnapshot();

    await page.type('#app > form > div:nth-child(5) > div > input', '2');
    expect(
      await page.evaluate(() => document.body.innerHTML),
    ).toMatchSnapshot();

    await page.waitForSelector('#app > form > div > div > select');
    await page.click('#app > form > div > div > select');
    expect(
      await page.evaluate(() => document.body.innerHTML),
    ).toMatchSnapshot();

    await page.select('#app > form > div > div > select', 'male');
    expect(
      await page.evaluate(() => document.body.innerHTML),
    ).toMatchSnapshot();

    await page.waitForSelector('#app > form > div:nth-child(9) > div > input');
    await page.click('#app > form > div:nth-child(9) > div > input');
    expect(
      await page.evaluate(() => document.body.innerHTML),
    ).toMatchSnapshot();

    await page.waitForSelector('body > #app > form > button');
    await page.click('body > #app > form > button');
    expect(
      await page.evaluate(() => document.body.innerHTML),
    ).toMatchSnapshot();

    await browser.close();
  });
});
