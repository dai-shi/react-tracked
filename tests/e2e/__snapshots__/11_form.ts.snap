// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`11_form > should work with recorded events 1`] = `
"
    <div id="app"><form>numRendered: 42<button type="submit" disabled="">Submit</button><h3>First Name</h3><div>numRendered: 44<div>First Name:<input value="">name is required</div></div><h3>Family Name</h3><div>numRendered: 46<div>Family Name:<input value="">name is required</div></div><h3>Gender</h3><div>numRendered: 38<div>Gender:<select><option value="na">N/A</option><option value="male">Male</option><option value="female">Female</option></select></div></div><h3>Teenager</h3><div>numRendered: 40<div>Teenager:<input type="checkbox" value="false"></div></div></form><div><a href="https://codesandbox.io/s/react-form-library-stress-test-4luf3">Stress Test</a></div></div>
  

"
`;

exports[`11_form > should work with recorded events 2`] = `
"
    <div id="app"><form>numRendered: 62<button type="submit" disabled="">Submit</button><h3>First Name</h3><div>numRendered: 64<div>First Name:<input value="1"></div></div><h3>Family Name</h3><div>numRendered: 46<div>Family Name:<input value="">name is required</div></div><h3>Gender</h3><div>numRendered: 38<div>Gender:<select><option value="na">N/A</option><option value="male">Male</option><option value="female">Female</option></select></div></div><h3>Teenager</h3><div>numRendered: 40<div>Teenager:<input type="checkbox" value="false"></div></div></form><div><a href="https://codesandbox.io/s/react-form-library-stress-test-4luf3">Stress Test</a></div></div>
  

"
`;

exports[`11_form > should work with recorded events 3`] = `
"
    <div id="app"><form>numRendered: 82<button type="submit" disabled="">Submit</button><h3>First Name</h3><div>numRendered: 84<div>First Name:<input value="11"></div></div><h3>Family Name</h3><div>numRendered: 46<div>Family Name:<input value="">name is required</div></div><h3>Gender</h3><div>numRendered: 38<div>Gender:<select><option value="na">N/A</option><option value="male">Male</option><option value="female">Female</option></select></div></div><h3>Teenager</h3><div>numRendered: 40<div>Teenager:<input type="checkbox" value="false"></div></div></form><div><a href="https://codesandbox.io/s/react-form-library-stress-test-4luf3">Stress Test</a></div></div>
  

"
`;

exports[`11_form > should work with recorded events 4`] = `
"
    <div id="app"><form>numRendered: 102<button type="submit" disabled="">Submit</button><h3>First Name</h3><div>numRendered: 104<div>First Name:<input value="111"></div></div><h3>Family Name</h3><div>numRendered: 46<div>Family Name:<input value="">name is required</div></div><h3>Gender</h3><div>numRendered: 38<div>Gender:<select><option value="na">N/A</option><option value="male">Male</option><option value="female">Female</option></select></div></div><h3>Teenager</h3><div>numRendered: 40<div>Teenager:<input type="checkbox" value="false"></div></div></form><div><a href="https://codesandbox.io/s/react-form-library-stress-test-4luf3">Stress Test</a></div></div>
  

"
`;

exports[`11_form > should work with recorded events 5`] = `
"
    <div id="app"><form>numRendered: 122<button type="submit">Submit</button><h3>First Name</h3><div>numRendered: 104<div>First Name:<input value="111"></div></div><h3>Family Name</h3><div>numRendered: 126<div>Family Name:<input value="2"></div></div><h3>Gender</h3><div>numRendered: 38<div>Gender:<select><option value="na">N/A</option><option value="male">Male</option><option value="female">Female</option></select></div></div><h3>Teenager</h3><div>numRendered: 40<div>Teenager:<input type="checkbox" value="false"></div></div></form><div><a href="https://codesandbox.io/s/react-form-library-stress-test-4luf3">Stress Test</a></div></div>
  

"
`;

exports[`11_form > should work with recorded events 6`] = `
"
    <div id="app"><form>numRendered: 142<button type="submit">Submit</button><h3>First Name</h3><div>numRendered: 104<div>First Name:<input value="111"></div></div><h3>Family Name</h3><div>numRendered: 146<div>Family Name:<input value="22"></div></div><h3>Gender</h3><div>numRendered: 38<div>Gender:<select><option value="na">N/A</option><option value="male">Male</option><option value="female">Female</option></select></div></div><h3>Teenager</h3><div>numRendered: 40<div>Teenager:<input type="checkbox" value="false"></div></div></form><div><a href="https://codesandbox.io/s/react-form-library-stress-test-4luf3">Stress Test</a></div></div>
  

"
`;

exports[`11_form > should work with recorded events 7`] = `
"
    <div id="app"><form>numRendered: 162<button type="submit">Submit</button><h3>First Name</h3><div>numRendered: 104<div>First Name:<input value="111"></div></div><h3>Family Name</h3><div>numRendered: 166<div>Family Name:<input value="222"></div></div><h3>Gender</h3><div>numRendered: 38<div>Gender:<select><option value="na">N/A</option><option value="male">Male</option><option value="female">Female</option></select></div></div><h3>Teenager</h3><div>numRendered: 40<div>Teenager:<input type="checkbox" value="false"></div></div></form><div><a href="https://codesandbox.io/s/react-form-library-stress-test-4luf3">Stress Test</a></div></div>
  

"
`;

exports[`11_form > should work with recorded events 8`] = `
"
    <div id="app"><form>numRendered: 162<button type="submit">Submit</button><h3>First Name</h3><div>numRendered: 104<div>First Name:<input value="111"></div></div><h3>Family Name</h3><div>numRendered: 166<div>Family Name:<input value="222"></div></div><h3>Gender</h3><div>numRendered: 38<div>Gender:<select><option value="na">N/A</option><option value="male">Male</option><option value="female">Female</option></select></div></div><h3>Teenager</h3><div>numRendered: 40<div>Teenager:<input type="checkbox" value="false"></div></div></form><div><a href="https://codesandbox.io/s/react-form-library-stress-test-4luf3">Stress Test</a></div></div>
  

"
`;

exports[`11_form > should work with recorded events 9`] = `
"
    <div id="app"><form>numRendered: 182<button type="submit">Submit</button><h3>First Name</h3><div>numRendered: 104<div>First Name:<input value="111"></div></div><h3>Family Name</h3><div>numRendered: 166<div>Family Name:<input value="222"></div></div><h3>Gender</h3><div>numRendered: 188<div>Gender:<select><option value="na">N/A</option><option value="male">Male</option><option value="female">Female</option></select></div></div><h3>Teenager</h3><div>numRendered: 40<div>Teenager:<input type="checkbox" value="false"></div></div></form><div><a href="https://codesandbox.io/s/react-form-library-stress-test-4luf3">Stress Test</a></div></div>
  

"
`;

exports[`11_form > should work with recorded events 10`] = `
"
    <div id="app"><form>numRendered: 202<button type="submit">Submit</button><h3>First Name</h3><div>numRendered: 104<div>First Name:<input value="111"></div></div><h3>Family Name</h3><div>numRendered: 166<div>Family Name:<input value="222"></div></div><h3>Gender</h3><div>numRendered: 188<div>Gender:<select><option value="na">N/A</option><option value="male">Male</option><option value="female">Female</option></select></div></div><h3>Teenager</h3><div>numRendered: 210<div>Teenager:<input type="checkbox" value="true"></div></div></form><div><a href="https://codesandbox.io/s/react-form-library-stress-test-4luf3">Stress Test</a></div></div>
  

"
`;

exports[`11_form > should work with recorded events 11`] = `
"
    <div id="app"><form>numRendered: 202<button type="submit">Submit</button><h3>First Name</h3><div>numRendered: 104<div>First Name:<input value="111"></div></div><h3>Family Name</h3><div>numRendered: 166<div>Family Name:<input value="222"></div></div><h3>Gender</h3><div>numRendered: 188<div>Gender:<select><option value="na">N/A</option><option value="male">Male</option><option value="female">Female</option></select></div></div><h3>Teenager</h3><div>numRendered: 210<div>Teenager:<input type="checkbox" value="true"></div></div></form><div><a href="https://codesandbox.io/s/react-form-library-stress-test-4luf3">Stress Test</a></div></div>
  

"
`;
